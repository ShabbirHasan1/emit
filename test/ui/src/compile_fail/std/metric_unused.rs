#![deny(unused_must_use)]

fn main() {
    let my_metric = 42;
    emit::metric!(value: my_metric);
}
