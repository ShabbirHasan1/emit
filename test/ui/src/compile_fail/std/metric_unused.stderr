error: unused return value of `__private_metric` that must be used
 --> src/compile_fail/std/metric_unused.rs:5:5
  |
5 |     emit::metric!(value: my_metric);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this macro returns a `Metric` without emitting it; sample it through an `emit::metric::Sampler`, or use the `emit::sample!` macro instead to sample and emit it
note: the lint level is defined here
 --> src/compile_fail/std/metric_unused.rs:1:9
  |
1 | #![deny(unused_must_use)]
  |         ^^^^^^^^^^^^^^^
help: use `let _ = ...` to ignore the resulting value
  |
5 |     let _ = emit::metric!(value: my_metric);
  |     +++++++
